### PR TITLE
give an uniform check for release process

### LINKED ISSUE #2538: Can it have necessary to give an Uniform Check for release name ?
I read the code of the release  processings, I find that in the processings such as Update, getstatus, gethistory and so on. Most of  those  processings no check the release name.

Can we have an Uniform Check for the release name ?  If the release name is illegal, processing exit and return the error message "The release name is illegal, release ""^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$"" and short than 53 chars.

The check is 

if !ValidName.MatchString(req.Name) {
		s.Log("uninstall: Release not found: %s", req.Name)
		return nil, errMissingRelease
	}

if len(req.Name) > releaseNameMaxLen {
		return nil, fmt.Errorf("release name %q exceeds max length of %d", req.Name, releaseNameMaxLen)
	}

### PR DESCRIPTION
**Pull request by yan234280533**
give an uniform check for release process ，fixed #2538

### CODE REVIEW

File: pkg/tiller/release_server.go [hunk 1]
```diff
@@ -359,3 +361,11 @@ func validateManifest(c environment.KubeClient, ns string, manifest []byte) erro
 	_, err := c.BuildUnstructured(ns, r)
 	return err
 }
+
+func validateReleaseName(releaseName string) error {
```
Comment (adamreese) [inline pkg/tiller/release_server.go]: This should return `errMissingRelease` if `releaseName == ""`

File: pkg/tiller/release_server_test.go [hunk 2]
```diff
@@ -197,8 +197,9 @@ func TestValidName(t *testing.T) {
 		" ":                      false,
 		".nina.":                 false,
 		"nina.pinta":             true,
+		"abcdefghi-abcdefghi-abcdefghi-abcdefghi-abcdefghi-abcd": false,
 	} {
-		if valid != ValidName.MatchString(name) {
+		if valid != (validateReleaseName(name) == nil) {
```
Comment (adamreese) [inline pkg/tiller/release_server_test.go]: This test could be updated to check against specific errors
```
for name, err := range map[string]error{
  "nina pinta santa-maria": errInvalidName,
  "nina-pinta-santa-maria": nil,
  "": errMissingRelease,
  ...
```

**Review by adamreese — CHANGES_REQUESTED** [review]

File: pkg/tiller/release_server_test.go (hunk 2 above)
Comment (yan234280533) [inline pkg/tiller/release_server_test.go]: @adamreese  Many thanks for you to review the code. I modified and fixed the code as you mentioned. Please help me to review the code, thanks a billion.

**Review by adamreese — APPROVED** [review]

File: pkg/tiller/release_server.go [hunk 3]
```diff
@@ -59,6 +59,8 @@ var (
 	errMissingRelease = errors.New("no release provided")
 	// errInvalidRevision indicates that an invalid release revision number was provided.
 	errInvalidRevision = errors.New("invalid release revision")
+	//errInvalidName indicates that an invalid release name was provided
+	errInvalidName = errors.New("invalid release name")
```
Comment (bacongobbler) [inline pkg/tiller/release_server.go]: any way we can return a more informative error as to why the release name is invalid? Something like "invalid release name. Must match regex \<insert ValidName regex here\>"

Comment (yan234280533) [inline pkg/tiller/release_server.go]: @bacongobbler  Many thanks for you review.  It a good idea, but   I worried is that it is return too long of the message.

Such as return  this error message:

errInvalidName = errors.New("invalid release name; must match  regex  ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$\  and the max length is 53 ")

Comment (yan234280533) [inline pkg/tiller/release_server.go]: @bacongobbler  I was modified the return message as which you suggested. Can you help me to review the commit again. So many thanks.

**Comment by technosophos** [comment]
Thanks! Merging now.

**Pull request merged**